bagelbids/tests/test_cli_pheno.py

```diff
@@ -1,27 +1,26 @@
 import json
 
-import pandas as pd
 import pytest
 from typer.testing import CliRunner
 
-from bagelbids import mappings
-from bagelbids.cli import (
-    are_not_missing,
-    bagel,
-    get_columns_about,
-    get_transformed_values,
-    is_missing_value,
-    map_categories_to_columns,
-    map_tools_to_columns,
-    transform_age,
-)
+from bagelbids.cli import bagel
 
 
 @pytest.fixture
 def runner():
     return CliRunner()
 
 
+@pytest.fixture
+def load_tmp_jsonld(tmp_path):
+    def _read_file():
```

Comment:
Aha. Yes, your understanding is correct. I did arrive at this configuration of the fixture after a bit of playing around with different versions of it (e.g., having the factory function itself accept a path arg, as you suggest), and chose the most concise option - which admittedly is probably not the most explicit. To answer your questions in reverse:

2. Actually, the only reason I wrote it this way **was** so that we wouldn't have to explicitly pass in `tmp_path` every time the fixture function is invoked in a test 🙂. But that's also a valid option:
```python
@pytest.fixture
def load_tmp_jsonld():
    def _read_file(temp_path):
        with open(temp_path / "pheno.jsonld", "r") as f:
            return json.load(f)

    return _read_file
```
works as well. I think you're right that this solution is clearer to the average reader - can make this change.  
1. From what I've read, the advantages of factory fixtures include them being able to be reused across modules without imports like you said; to use other fixtures themselves with tests being agnostic to the extra dependency injection (e.g., if `read_file` were a regular function but depended on the `test_data` fixture (as the version of this fixture in test_utility.py does), how would a test that calls `read_file` handle the fixture?) and be used by other fixtures themselves; and readability like other fixtures, since all the required initializations for each test can be found directly in the test definition/signature.